platform/o.n.bootstrap/test/unit/src/org/netbeans/ModuleManagerTest.java

```diff
@@ -2760,6 +2760,189 @@ public void testMissingSpecVersion() throws Exception {
         Module client = mgr.create(jar, null, false, false, false);
         assertEquals(1, client.getProblems().size());
     }
+
+    public void testEnableHostWithDefautFragment() throws Exception {
+        MockModuleInstaller installer = new MockModuleInstaller();
+        MockEvents ev = new MockEvents();
+        ModuleManager mgr = new ModuleManager(installer, ev);
+        mgr.mutexPrivileged().enterWriteAccess();
+
+        Module host = mgr.create(new File(jars, "host-module.jar"), null, false, false, false);
+        Module fragment = mgr.create(new File(jars, "fragment-module.jar"), null, false, false, false);
+
+        assertTrue("Host is known", mgr.getModules().contains(host));
+        assertTrue("Fragment is known", mgr.getModules().contains(fragment));
+
+        mgr.enable(new HashSet<>(Arrays.asList(host, fragment)));
+
+        assertTrue("Host must be enabled", mgr.getEnabledModules().contains(host));
+        assertTrue("Fragment must be enabled", mgr.getEnabledModules().contains(fragment));
+    }
+
+    public void testEnableHostWithDefautFragmentUnsatisfied() throws Exception {
+        MockModuleInstaller installer = new MockModuleInstaller();
+        MockEvents ev = new MockEvents();
+        ModuleManager mgr = new ModuleManager(installer, ev);
+        mgr.mutexPrivileged().enterWriteAccess();
+
+        createTestJAR(data, jars, "fragment-module-missing-token", null);
+
+        Module host = mgr.create(new File(jars, "host-module.jar"), null, false, false, false);
+        Module fragment = mgr.create(new File(jars, "fragment-module-missing-token.jar"), null, false, false, false);
+
+        assertTrue("Host is known", mgr.getModules().contains(host));
+        assertTrue("Fragment is known", mgr.getModules().contains(fragment));
+
+        boolean expectedExceptionCatched = false;
+        try {
+            mgr.enable(fragment);
+        } catch (IllegalArgumentException e) {
+            expectedExceptionCatched = e.getMessage().startsWith("Not all requested modules can be enabled:");
```

Comment:
Would it be possible to check for `contains` with the failed fragment's module name ?